File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-2.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int l = 0;
        int r = 0;

        int n = nums.size();

        while (r < n) {
            nums[l] = nums[r];

            while((r < n) && nums[l] == nums[r]){
                r+=1;
            }
            l+=1;
        }

        return l;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        seen = set(nums)
        max_length = 0

        for num in seen:
            # expand only if num is the start
            if (num - 1) not in seen:
                length = 1
                while num + length in seen:
                    length += 1
                max_length = max(max_length, length)

        return max_length
        